docs/source/evaluation_scripts.md

```diff
@@ -129,3 +129,205 @@ Here are the steps to configure remote evaluation:
     The `evaluate()` method also accepts keyword arguments.
 
     **IMPORTANT** ⚠️: If the `evaluate()` method fails due to any reason or there is a problem with the submission, please ensure to raise an `Exception` with an appropriate message.
+
+### Writing Code Upload Challenge Evaluation
+
+Each challenge has an evaluation script, which evaluates the submission of participants and returns the scores which will populate the leaderboard. The logic for evaluating and judging a submission is customizable and varies from challenge to challenge, but the overall structure of evaluation scripts is fixed due to architectural reasons.
+
+In code-upload challenges, the evaluation is tighly-coupled with the agent and environment containers: The agent container continuously interacts with the environment container, and the environment container provides feedback to the agent until the environment provides a `stop` signal when the goal is reached or when the agent cannot continue further. Therefore, it is important to describe the interaction between the containers, how once can set up the individual containers, and how can the evaluation be performed.
```

Comment:
>to the agent until the environment provides a `stop` signal when the goal is reached or when the agent cannot continue further

The environment doesn't provide a stop signal. The agent outputs a stop action for each episode. The environment asks agents for actions until all episodes run out or the time limit for evaluation is reached

Reply:
Fixed. Added the above lines.